### DISCUSSION TITLE
[Technical Question] sendReportedProperties where property value has custom Jackson annotations

### DISCUSSION THREAD

**Discussion by jokarl** [Azure/azure-iot-sdk-java#1269] [General]
In our project we rely heavily on Jackson annotations and custom deserializers. When I want to report one of our objects using `sendReportedProperties`, things get a bit iffy:

```java
deviceClient.sendReportedProperties(Set.of(new Property("config", OurJacksonAnnotatedClass.builder().withPosition(Position.from(0,0)).build())));
```

This updates  the twin but serializes a lot of internals of the class that we don't want. I could serialize the object myself using Jackson, but that will instead generate an escape JSON string as the value, when I actually want an object.

Is there any way to override GSON in favor of Jackson, or any other way to handle this scenario?

AB#8306228

**Comment by timtay-microsoft** [comment]
Currently our SDK only supports Gson based json objects, and there is no configuration that we enable. We'll take this as a feature request though.
